sample/src/main/java/com/vimeo/sample/network/NetworkRequest.java

```diff
@@ -116,16 +120,17 @@ void cancelRequest() {
                     builder.append(line);
                 }
 
+                Stag.Factory factory = new Stag.Factory();
                 Gson gson = new GsonBuilder().registerTypeAdapter(Date.class, new DateParser())
-                        .registerTypeAdapterFactory(new Stag.Factory())
+                        .registerTypeAdapterFactory(factory)
                         .create();
 
                 JsonObject jsonObject = new JsonParser().parse(builder.toString()).getAsJsonObject();
 
+                TypeAdapter<VideoList> videoListTypeAdapter = factory.getVideoList$TypeAdapter(gson);
                 long time = System.currentTimeMillis();
 
-                videos.addAll(gson.fromJson(jsonObject, VideoList.class).data);
-
+                videos.addAll(videoListTypeAdapter.fromJsonTree(jsonObject).data);
```

Comment:
While this works, do you think it would more helpful if we added an API method to the generated `Stag` class:

```java
class Stag {
    public <T> T fromJson(Gson gson, JsonObject jsonObject, Class<T> clazz) {
        switch (clazz.getName()) {
            case "com.vimeo.sample.model.VideoList":
                return getVideoList$TypeAdapter(gson).fromJsonTree(jsonObject);
        }
    }
}

...
videos.addAll(factory.fromGson(gson, jsonObject, VideoList.class));
```

that way it's the same number of lines as before + the extra line to declare the factory object on its own line. This would be a little cumbersome and redundant to generate though since it would look very similar to the `Factory.create` method.